Documents/element_design.md

```diff
@@ -0,0 +1,46 @@
+- **Consider Events:** Think about what events the element should respond to, e.g., onClick, onFocus.
+
+### 3: Consider Inheritance and Abstraction
+- **Base Classes:** Look for any existing elements in JDI-Light that are similar and can act as base classes. This helps in reusing code and keeping the design DRY (Don't Repeat Yourself).
+- **Abstract Elements:** If needed, design an abstract class that holds common properties and methods for a category of elements. For example, all form controls might share certain behaviors.
+
+### 4: Incorporate Element Locators
+- **Locators:** Define how the element will be located in the DOM. Choose the most stable locators to ensure reliable tests.
+- **Methods for Locators:** Decide how the element can change its locators if needed, to make the custom element flexible and reusable in different contexts.
+
+### 5: Design for extensibility
+- Consider how the element can be customized or extended. Users of your framework might need to adjust the default behavior or appearance.
+
+### 6: Mockups and Prototypes
+- **Visual design:** Draw a visual representation of the element to understand how it should look and behave. Use prototyping platforms like Figma, Sketch, or InVision.
+- **Prototype:** Define example scenarios where the UI element would be used, what the expected user input and system responses would be. This can include common and edge case scenarios. Write detailed UI specifications that describe the functionality, appearance, and behavior of the element. These should include dimensions, color schemes, typographic details, and states (e.g., hover, disabled).
+
+### 7: Document the Design
+- **Documentation:** Write clear documentation outlining how the element is supposed to work and how it should be used within the JDI-Light framework.
+- **Examples:** Provide examples of usage including variations of the element if applicable.
+- **Javadoc** Write all documentation related to an element in a form of Javadoc inside an element class.
+- **Task for frontend** Express all abowementioned actions (in 6 and 7 steps) in a form of a task for a frontend developer in order to adjust testsite - to get element you designed on it.
+
+### 8: Plan for Testing
+- **Testing Strategy:** Determine how you will test the custom element within the JDI-Light framework.
+- **Test Cases:** Outline test cases to cover all functionalities and states of the element. Keep in mind that you need to represent all features of your element.
+
+
+### Notes:
+- Make sure that locators you using in a framework is not implementation-specific. You must use locators which will not change on a different web-site implementation. If you are unable to present your element w/o usage implementation-specific locator - create a custom element inside a test project and use it in test.
+- Consider naming conventions that match the JDI-Light framework to maintain coherence and reduce the learning curve for users of your custom library.
```

Comment:
I think it is good to add a link to some good example and also mention that we have checkstyle enabled

Reply:
added